packages/types/index.ts

```diff
@@ -67,7 +67,7 @@ export interface ConfigObject {
   /** Globs of files to ignore */
   ignores?: Array<string>
   /** Severity of rules and individual rule options. */
-  rules?: {
+  rules: {
```

Comment:
issue: omitting `rules` is a way of saying "apply this to all rules. It's made to let users quickly ignore certain parts of their codebase like tests or compiled files

suggestion: let's make this key optional and add support for these cases

Reply:
I would be happy to but if we don't have `rules` in a config object, it means that we have only globs, and then there is nothing to apply. 

We could think up an alternative syntax or extend the object shape for that purpose.

E.g. I thought of the following solutions to the issue

1. You still need to specify the `rules` object but can apply severities to all rules
```javascript
export default defineConfig([
  ...fsd.configs.recommended,
  {
    files: ['**/shared/**'],
    rules: {
      '*': 'off' // can only set severity in this case (options obviously cannot be set)
    }
  }
])
```

2. Additional "severity" field that is processed only when the rules object is absent (otherwise it's skipped). But this slightly contradicts with the declared policy of keeping the public interface as concise as possible.
```javascript
export default defineConfig([
  ...fsd.configs.recommended,
  {
    files: ['**/shared/**'],
    severity: 'off'
  }
])
```

Reply:
Ah, you're right, I didn't think this through. I just kind of copied ESLint here without really thinking if it makes sense or not.

In ESLint, there's a [special kind of configuration object](https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores) — if you only specify the `ignores` key and nothing else, that will be taken to mean "ignore these files for all rules". That's not how `ignores` behaves when you have `files` or `rules` specified, but I think I'd prefer consistency with ESLint here over creating a new syntax. 

So I think that our `ConfigObject` should be `{ files?, ignores?, rules } | { ignores }`, so then we don't make `rules` optional, but still leave an opportunity to only have `ignores` and nothing else

Reply:
No problem, I can implement logic handling that `{ ignores: [] }` case.

There’s no doubt that ESLint is an amazing high-quality tool, one of the best of its type. I have been using it myself for many years. And since we're developing a tool of the same type, references to ESLint are 100% reasonable, and making the config structure similar to it should allow people to onboard to Steiger quicker and smoother.

I just think that we would like to be sure that we consider what's the best/more convenient for users in our particular case in the first place and we don't just blindly copy and paste from ESLint/Stylelint/… whatever.